flox/core.py

```diff
@@ -589,28 +609,39 @@ def chunk_reduce(
     dict
     """
 
-    if dtype is not None:
-        assert isinstance(dtype, Sequence)
-    if fill_value is not None:
-        assert isinstance(fill_value, Sequence)
-
-    if isinstance(func, str) or callable(func):
-        func = (func,)  # type: ignore
+    if not (isinstance(func, str) or callable(func)):
+        funcs = func
+    else:
+        funcs = (func,)
+    nfuncs = len(funcs)
 
-    func: Sequence[str] | Sequence[Callable]
+    if isinstance(dtype, Sequence):
+        dtypes = dtype
+    else:
+        dtypes = (dtype,) * nfuncs
+    assert len(dtypes) >= nfuncs
 
-    nax = len(axis) if isinstance(axis, Sequence) else by.ndim
-    final_array_shape = array.shape[:-nax] + (1,) * (nax - 1)
-    final_groups_shape = (1,) * (nax - 1)
+    if isinstance(fill_value, Sequence):
+        fill_values = fill_value
+    else:
+        fill_values = (fill_value,) * nfuncs
+    assert len(fill_values) >= nfuncs
 
-    if isinstance(axis, Sequence) and len(axis) == 1:
-        axis = next(iter(axis))
+    if isinstance(kwargs, Sequence):
+        kwargss = kwargs
+    else:
+        kwargss = ({},) * nfuncs
+    assert len(kwargss) >= nfuncs
 
-    if not isinstance(fill_value, Sequence):
-        fill_value = (fill_value,)
+    if isinstance(axis, Sequence):
+        nax = len(axis)
+        if nax == 1:
```

Comment:
is this `if` necessary?

Reply:
Not sure. It's reformatted from the original code:
```python
if isinstance(axis, Sequence) and len(axis) == 1:
        axis = next(iter(axis))
```

So I guess I'll return the question? :p
Should we even allow sequences of axis if we only use the first value anyway?

Reply:
OK it should be -1 always at this point. It isn't actually used later. Let's just keep it.